torch/distributed/fsdp/flat_param.py

```diff
@@ -1055,7 +1063,7 @@ def pre_unshard(self) -> bool:
         matches the dtype of the expected unsharded parameter.
         """
         ret = False
-        if self._use_orig_params:
+        if self._use_orig_params and not self._skipped_use_sharded_views:
```

Comment:
does this mean writeback feature gets disabled for zero2? can we just call use_sharded_views here?